src/lib/telemetry/TelemetryReporter.ts

```diff
@@ -117,16 +140,44 @@ export class TelemetryReporter {
       }
       return error
     })
-    await ensureFile(metaErrorFilePath)
-    await writeJson(metaErrorFilePath, { errors: errorsReport })
+    await ensureDir(join(TelemetryReporter.PENDING_DATA_DIR, 'errors'))
+    await writeJson(metaErrorFilePath, errorsReport)
   }
 
-  private async pendingFilesPaths() {
-    const pendingDataFiles = (await readdir(TelemetryReporter.PENDING_DATA_DIR)).filter(fileName => {
-      return fileName !== this.dataPendingLock.lockName
-    })
+  public async ensureTelemetryDirMaxSize() {
+    const telemetryFiles = await glob('**', { cwd: TelemetryCollector.TELEMETRY_LOCAL_DIR, absolute: true })
+    let telemetryDirSize = 0
+    await Promise.all(
+      telemetryFiles.map(async file => {
+        const fileStats = await stat(file)
+        telemetryDirSize += fileStats.size
+      })
+    )
+    if (telemetryDirSize > TelemetryReporter.MAX_TELEMETRY_DIR_SIZE) {
+      await remove(TelemetryCollector.TELEMETRY_LOCAL_DIR)
+    }
+  }
+
+  private async treatReportError(reportError, reportType: 'metrics', metricsOrErrors: MetricReport[])
```

Comment:
Arguments names doesn't have to have the same name from the function when overloading
```suggestion
  private async treatReportError(reportError, reportType: 'metrics', metrics: MetricReport[])
```